gpt4all-chat/database.cpp

```diff
@@ -0,0 +1,1072 @@
+#include "database.h"
+#include "download.h"
+
+#include <QTimer>
+#include <QPdfDocument>
+
+//#define DEBUG
+//#define DEBUG_EXAMPLE
+
+#define LOCALDOCS_VERSION 0
+
+const auto INSERT_CHUNK_SQL = QLatin1String(R"(
```

Comment:
All of these, as well as the helper functions that are only called from database.cpp, should be made static or put inside an anonymous namespace. (If they actually need to be externally visible for some reason I didn't realize then never mind, but to me it looks like they don't.)

Reply:
They aren't externally visible as they are defined inside an impl file.